src/main/java/net/dries007/tfc/util/agriculture/Crop.java

```diff
@@ -123,59 +181,44 @@ else if (currentStage == getMaxStage() - 1)
         return ItemStack.EMPTY;
     }
 
-    public BlockCropTFC create()
+    public BlockCropTFC createGrowingBlock()
     {
         if (type == SIMPLE || type == PICKABLE)
         {
-            switch (growthStages)
-            {
-                case 5:
-                    return new BlockCropSimple(this, type == PICKABLE)
-                    {
-                        @Override
-                        public PropertyInteger getStageProperty()
-                        {
-                            return STAGE_5;
-                        }
-                    };
-                case 6:
-                    return new BlockCropSimple(this, type == PICKABLE)
-                    {
-                        @Override
-                        public PropertyInteger getStageProperty()
-                        {
-                            return STAGE_6;
-                        }
-                    };
-                case 7:
-                    return new BlockCropSimple(this, type == PICKABLE)
-                    {
-                        @Override
-                        public PropertyInteger getStageProperty()
-                        {
-                            return STAGE_7;
-                        }
-                    };
-                case 8:
-                    return new BlockCropSimple(this, type == PICKABLE)
-                    {
-                        @Override
-                        public PropertyInteger getStageProperty()
-                        {
-                            return STAGE_8;
-                        }
-                    };
-            }
+            return BlockCropSimple.create(this, type == PICKABLE);
         }
         else if (type == SPREADING)
         {
-            return new BlockCropSpreading(this);
+            return BlockCropSpreading.create(this);
         }
         throw new IllegalStateException("Invalid growthstage property " + growthStages + " for crop");
     }
 
+    public BlockCropDead createDeadBlock()
+    {
+        return new BlockCropDead(this);
+    }
+
     enum CropType
     {
         SIMPLE, PICKABLE, SPREADING
     }
+
+    private static class Range
+    {
+        public final float minAlive, maxAlive, minGrow, maxGrow;
+
+        public Range(float minGrow, float maxGrow, float tolerance)
+        {
+            this(minGrow - tolerance, minGrow, maxGrow, maxGrow + tolerance);
+        }
+
+        public Range(float minAlive, float minGrow, float maxGrow, float maxAlive)
+        {
+            this.minAlive = minAlive;
+            this.minGrow = minGrow;
+            this.maxGrow = maxGrow;
+            this.maxAlive = maxAlive;
+        }
+    }
```

Comment:
This is rather pointless abstraction, it's not used anywhere, and the entire point seems to avoid a calculation which could just as easily be done in constructor arguments for `Crop` instead.

Reply:
I thought it would be useful for making the constructors more clear and readable, rather than a series of a bunch of numbers in the arguments on all one line, which is also why I opted to add newlines at designated points in the code, helps readability...

Reply:
These constructors are auto-generated - there's no use in making them more readable as it's much easier to look and modify a spreadsheet instead.

Reply:
Range has been removed.